R/shiny_models_reg.R

```diff
@@ -0,0 +1,122 @@
+#'  [shiny_models()] for an object of `reg_shiny_data` class
+#'
+#' @export
+#' @rdname shiny_models
+shiny_models.reg_shiny_data <-
+  function(x,
+           hover_cols = NULL,
+           hover_only = FALSE,
+           ...) {
+    preds <- x$predictions
+    num_columns <- x$num_cols
+    fac_columns <- x$fac_cols
+    ui <- shinydashboard::dashboardPage(
+      shinydashboard::dashboardHeader(title = "Shinymodels"),
+      shinydashboard::dashboardSidebar(
+        shinydashboard::sidebarMenu(
+          shinydashboard::menuItem("Plots", tabName = "interactive", icon = icon("chart-line")),
+          shiny::helpText("Select column(s) to create plots"),
+          if (length(num_columns) == 0) {
+            shiny::helpText("No numeric column to display")
+          }
+          else {
+            shiny::selectInput(
+              inputId = "num_value_col",
+              label = "Numeric Columns",
+              choices = unique(c("None Selected" = "", num_columns))
+            )
+          },
+          if (length(fac_columns) == 0) {
+            shiny::helpText("No factor column to display")
+          }
+          else {
+            shiny::selectInput(
+              inputId = "factor_value_col",
+              label = "Factor Columns",
+              choices = unique(c("None Selected" = "", fac_columns))
+            )
+          },
+          shiny::helpText("Select the opacity of the points"),
+          # Input: Simple integer interval ----
+          shiny::sliderInput("alpha", "Alpha:",
+                             min = 0, max = 1,
+                             value = 0.7, step = 0.1
+          ),
+          shiny::helpText("Select the size of the points"),
+          # Input: Simple integer interval ----
+          shiny::sliderInput("size", "Size:",
+                             min = 0.5, max = 3,
+                             value = 1.5, step = 0.5
+          )
+        )
+      ),
+      shinydashboard::dashboardBody(
+        shinydashboard::tabItems(
+          # First tab content
+          shinydashboard::tabItem(
+            tabName = "interactive",
+            shiny::fluidRow(
+              boxed(plotly::plotlyOutput("obs_vs_pred"),
+                    "Observed vs. Predicted"),
+              boxed(plotly::plotlyOutput("resid_vs_pred"),  "Residuals vs Predicted"),
+              if (length(num_columns) == 0) {
```

Comment:
You should not need the `if/else` branching. The `input` argument to `boxed` takes care of this.